src/main/java/com/purbon/kafka/topology/JulieOps.java

```diff
@@ -219,25 +219,29 @@ static void verifyRequiredParameters(String topologyFile, Map<String, String> co
     }
   }
 
-  void run(ExecutionPlan plan) throws IOException {
+  void buildAndExecutePlan(BackendController backendController, PrintStream printStream)
+      throws IOException {
+    ExecutionPlan plan = ExecutionPlan.init(backendController, printStream);
     LOGGER.debug(
         String.format(
-            "Running topology builder with TopicManager=[%s], accessControlManager=[%s], dryRun=[%s], isQuite=[%s]",
+            "Running topology builder with topicManager=[%s], accessControlManager=[%s], dryRun=[%s], isQuiet=[%s]",
             topicManager, accessControlManager, config.isDryRun(), config.isQuiet()));
 
-    // Create users should always be first, so user exists when making acl link
-    principalManager.applyCreate(topology, plan);
-
-    topicManager.apply(topology, plan);
-    accessControlManager.apply(topology, plan);
-
-    connectorManager.apply(topology, plan);
-    kSqlArtefactManager.apply(topology, plan);
-
-    // Delete users should always be last,
-    // avoids any unlinked acls, e.g. if acl delete or something errors then there is a link still
-    // from the account, and can be re-run or manually fixed more easily
-    principalManager.applyDelete(topology, plan);
+    ExecutionPlanUpdater[] updaters =
+        new ExecutionPlanUpdater[] {
+          principalManager,
+          topicManager,
+          accessControlManager,
+          connectorManager,
+          kSqlArtefactManager
+        };
+
+    for (ExecutionPlanUpdater updater : updaters) {
+      updater.updatePlan(plan, topology);
+    }
+    for (ExecutionPlanUpdater updater : updaters) {
+      updater.updatePlanWithFinalActions(plan, topology);
```

Comment:
Betre å splitte principalManager i to managers og unngå denne metoden.

Reply:
Done.